shared/router-v2/router.desktop.tsx

```diff
@@ -38,121 +38,219 @@ import OutOfDate from '../app/out-of-date'
+  const getNavigationOptions: undefined | Object | ((n: {navigation: typeof childNav}) => Object) =
+    // @ts-ignore
+    Component?.navigationOptions
+  let navigationOptions: undefined | Object
+  if (typeof getNavigationOptions === 'function') {
+    navigationOptions = getNavigationOptions({navigation: childNav})
+  } else if (typeof getNavigationOptions === 'object') {
+    navigationOptions = getNavigationOptions
   }
-}
-
-class ModalView extends React.PureComponent<NavigationViewProps<any>> {
-  render() {
-    const navigation = this.props.navigation
-    const index = navigation.state.index
-    const activeKey = navigation.state.routes[index].key
-    const descriptor = this.props.descriptors[activeKey]
-    const childNav = descriptor.navigation
-
-    // We render the app below us
-    const appKey = this.props.navigation.state.routes[0].key
-    const appNav = this.props.navigation.getChildNavigation(appKey)
-    const appDescriptor = this.props.descriptors[appKey]
-
-    // TODO we might want all the click handling / grey background stuff handled by the routing
-    return (
-      <>
-        <SceneView
-          key="AppLayer"
-          navigation={appNav}
-          component={appDescriptor.getComponent()}
-          screenProps={this.props.screenProps || noScreenProps}
-        />
-        {index > 0 && (
-          <Kb.Box2 direction="vertical" style={styles.modalContainer}>
-            <SceneView
-              key="ModalLayer"
-              navigation={childNav}
-              component={descriptor.getComponent()}
-              screenProps={this.props.screenProps || noScreenProps}
-            />
+  // @ts-ignore
+  const {modal2Style, modal2AvoidTabs, modal2, modal2ClearCover, modal2Type} = navigationOptions ?? {}
+
+  const popRef = React.useRef(navigation.pop)
+  React.useEffect(() => {
+    popRef.current = navigation.pop
+  }, [navigation])
+
+  const backgroundRef = React.useRef(null)
+  // we keep track of mouse down/up to determine if we should call it a 'click'. We don't want dragging the
+  // window around to count
+  const [mouseDownX, setMouseDownX] = React.useState(mouseResetValue)
+  const [mouseDownY, setMouseDownY] = React.useState(mouseResetValue)
+  const onMouseDown = React.useCallback(
+    (e: React.MouseEvent) => {
+      const {screenX, screenY, target} = e.nativeEvent
+      if (target !== backgroundRef.current) {
+        return
+      }
+      setMouseDownX(screenX)
+      setMouseDownY(screenY)
+    },
+    [setMouseDownX, setMouseDownY]
+  )
+  const onMouseUp = React.useCallback(
+    (e: React.MouseEvent) => {
+      const {screenX, screenY, target} = e.nativeEvent
+      if (target !== backgroundRef.current) {
+        return
+      }
+      const delta = Math.abs(screenX - mouseDownX) + Math.abs(screenY - mouseDownY)
+      const dismiss = delta < mouseDistanceThreshold
+      setMouseDownX(mouseResetValue)
+      setMouseDownY(mouseResetValue)
+      if (dismiss) {
```

Comment:
seems to be something up with this, clicking above or below the modal dismisses, but to the left or right doesn't,

Reply:
ah i changed that recently, will fix tnx